<!DOCTYPE html>
<html>
  <head>
    <title>selenium.common.exceptions.WebDriverException: Message: 'chromedriver' executable needs to be in PATH. Please see https://chromedriver.chromium.org/home

 // Werkzeug Debugger</title>
    <link rel="stylesheet" href="?__debugger__=yes&amp;cmd=resource&amp;f=style.css">
    <link rel="shortcut icon"
        href="?__debugger__=yes&amp;cmd=resource&amp;f=console.png">
    <script src="?__debugger__=yes&amp;cmd=resource&amp;f=debugger.js"></script>
    <script>
      var CONSOLE_MODE = false,
          EVALEX = true,
          EVALEX_TRUSTED = false,
          SECRET = "QgHs7Q5SWH7qg5HgJ7Jc";
    </script>
  </head>
  <body style="background-color: #fff">
    <div class="debugger">
<h1>WebDriverException</h1>
<div class="detail">
  <p class="errormsg">selenium.common.exceptions.WebDriverException: Message: &#x27;chromedriver&#x27; executable needs to be in PATH. Please see https://chromedriver.chromium.org/home

</p>
</div>
<h2 class="traceback">Traceback <em>(most recent call last)</em></h2>
<div class="traceback">
  <h3></h3>
  <ul><li><div class="frame" id="frame-140636034197536">
  <h4>File <cite class="filename">"/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/common/service.py"</cite>,
      line <em class="line">71</em>,
      in <code class="function">start</code></h4>
  <div class="source "><pre class="line before"><span class="ws">           </span>or when it can&#x27;t connect to the service</pre>
<pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>cmd = [self.path]</pre>
<pre class="line before"><span class="ws">            </span>cmd.extend(self.command_line_args())</pre>
<pre class="line current"><span class="ws">            </span>self.process = subprocess.Popen(cmd, env=self.env,</pre>
<pre class="line after"><span class="ws">                                            </span>close_fds=system() != &#x27;Windows&#x27;,</pre>
<pre class="line after"><span class="ws">                                            </span>stdout=self.log_file,</pre>
<pre class="line after"><span class="ws">                                            </span>stderr=self.log_file,</pre>
<pre class="line after"><span class="ws">                                            </span>stdin=PIPE,</pre>
<pre class="line after"><span class="ws">                                            </span>creationflags=self.creationflags)</pre></div>
</div>

<li><div class="frame" id="frame-140636034197760">
  <h4>File <cite class="filename">"/usr/lib/python3.9/subprocess.py"</cite>,
      line <em class="line">951</em>,
      in <code class="function">__init__</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">                </span>self.stderr = io.open(errread, &#x27;rb&#x27;, bufsize)</pre>
<pre class="line before"><span class="ws">                </span>if self.text_mode:</pre>
<pre class="line before"><span class="ws">                    </span>self.stderr = io.TextIOWrapper(self.stderr,</pre>
<pre class="line before"><span class="ws">                            </span>encoding=encoding, errors=errors)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line current"><span class="ws">            </span>self._execute_child(args, executable, preexec_fn, close_fds,</pre>
<pre class="line after"><span class="ws">                                </span>pass_fds, cwd, env,</pre>
<pre class="line after"><span class="ws">                                </span>startupinfo, creationflags, shell,</pre>
<pre class="line after"><span class="ws">                                </span>p2cread, p2cwrite,</pre>
<pre class="line after"><span class="ws">                                </span>c2pread, c2pwrite,</pre>
<pre class="line after"><span class="ws">                                </span>errread, errwrite,</pre></div>
</div>

<li><div class="frame" id="frame-140636034197984">
  <h4>File <cite class="filename">"/usr/lib/python3.9/subprocess.py"</cite>,
      line <em class="line">1821</em>,
      in <code class="function">_execute_child</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">                        </span>err_filename = cwd</pre>
<pre class="line before"><span class="ws">                    </span>else:</pre>
<pre class="line before"><span class="ws">                        </span>err_filename = orig_executable</pre>
<pre class="line before"><span class="ws">                    </span>if errno_num != 0:</pre>
<pre class="line before"><span class="ws">                        </span>err_msg = os.strerror(errno_num)</pre>
<pre class="line current"><span class="ws">                    </span>raise child_exception_type(errno_num, err_msg, err_filename)</pre>
<pre class="line after"><span class="ws">                </span>raise child_exception_type(err_msg)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>def _handle_exitstatus(self, sts,</pre>
<pre class="line after"><span class="ws">                               </span>waitstatus_to_exitcode=os.waitstatus_to_exitcode,</pre></div>
</div>

<li><div class="exc-divider">During handling of the above exception, another exception occurred:</div>
<li><div class="frame" id="frame-140636034196080">
  <h4>File <cite class="filename">"/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py"</cite>,
      line <em class="line">2091</em>,
      in <code class="function">__call__</code></h4>
  <div class="source "><pre class="line before"><span class="ws">    </span>def __call__(self, environ: dict, start_response: t.Callable) -&gt; t.Any:</pre>
<pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;The WSGI server calls the Flask application object as the</pre>
<pre class="line before"><span class="ws">        </span>WSGI application. This calls :meth:`wsgi_app`, which can be</pre>
<pre class="line before"><span class="ws">        </span>wrapped to apply middleware.</pre>
<pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
<pre class="line current"><span class="ws">        </span>return self.wsgi_app(environ, start_response)</pre></div>
</div>

<li><div class="frame" id="frame-140636034196192">
  <h4>File <cite class="filename">"/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py"</cite>,
      line <em class="line">2076</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source "><pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line before"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
<pre class="line before"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line before"><span class="ws">                </span>error = e</pre>
<pre class="line current"><span class="ws">                </span>response = self.handle_exception(e)</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre>
<pre class="line after"><span class="ws">                </span>raise</pre>
<pre class="line after"><span class="ws">            </span>return response(environ, start_response)</pre>
<pre class="line after"><span class="ws">        </span>finally:</pre></div>
</div>

<li><div class="frame" id="frame-140636034196416">
  <h4>File <cite class="filename">"/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py"</cite>,
      line <em class="line">2073</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source "><pre class="line before"><span class="ws">        </span>ctx = self.request_context(environ)</pre>
<pre class="line before"><span class="ws">        </span>error: t.Optional[BaseException] = None</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line current"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
<pre class="line after"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">                </span>error = e</pre>
<pre class="line after"><span class="ws">                </span>response = self.handle_exception(e)</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre></div>
</div>

<li><div class="frame" id="frame-140636034196528">
  <h4>File <cite class="filename">"/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py"</cite>,
      line <em class="line">1518</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source "><pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line before"><span class="ws">                </span>rv = self.dispatch_request()</pre>
<pre class="line before"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line current"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(</pre>
<pre class="line after"><span class="ws">        </span>self,</pre>
<pre class="line after"><span class="ws">        </span>rv: t.Union[ResponseReturnValue, HTTPException],</pre></div>
</div>

<li><div class="frame" id="frame-140636034196640">
  <h4>File <cite class="filename">"/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py"</cite>,
      line <em class="line">1516</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source "><pre class="line before"><span class="ws">        </span>self.try_trigger_before_first_request_functions()</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line current"><span class="ws">                </span>rv = self.dispatch_request()</pre>
<pre class="line after"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(</pre></div>
</div>

<li><div class="frame" id="frame-140636034196752">
  <h4>File <cite class="filename">"/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py"</cite>,
      line <em class="line">1502</em>,
      in <code class="function">dispatch_request</code></h4>
  <div class="source "><pre class="line before"><span class="ws">            </span>getattr(rule, &quot;provide_automatic_options&quot;, False)</pre>
<pre class="line before"><span class="ws">            </span>and req.method == &quot;OPTIONS&quot;</pre>
<pre class="line before"><span class="ws">        </span>):</pre>
<pre class="line before"><span class="ws">            </span>return self.make_default_options_response()</pre>
<pre class="line before"><span class="ws">        </span># otherwise dispatch to the handler for that endpoint</pre>
<pre class="line current"><span class="ws">        </span>return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def full_dispatch_request(self) -&gt; Response:</pre>
<pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;Dispatches the request and on top of that performs request</pre>
<pre class="line after"><span class="ws">        </span>pre and postprocessing as well as HTTP exception catching and</pre>
<pre class="line after"><span class="ws">        </span>error handling.</pre></div>
</div>

<li><div class="frame" id="frame-140636034196864">
  <h4>File <cite class="filename">"/home/lubuntu/cars_project/carsandbids_project/carsandbids/carsandbids/selenium_api.py"</cite>,
      line <em class="line">107</em>,
      in <code class="function">page</code></h4>
  <div class="source "><pre class="line before"><span class="ws">    </span>return get_new_api(url)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws"></span>@app.route(&#x27;/page&#x27;,methods=[&quot;POST&quot;,&quot;GET&quot;])</pre>
<pre class="line before"><span class="ws"></span>def page():</pre>
<pre class="line before"><span class="ws">    </span>url = str(request.args.get(&quot;url&quot;))</pre>
<pre class="line current"><span class="ws">    </span>return get_page(url)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span>@app.route(&#x27;/past&#x27;,methods=[&quot;POST&quot;,&quot;GET&quot;])</pre>
<pre class="line after"><span class="ws"></span>def past():</pre>
<pre class="line after"><span class="ws">    </span>url = str(request.args.get(&quot;url&quot;))</pre>
<pre class="line after"><span class="ws">    </span>return get_past_api(url)</pre></div>
</div>

<li><div class="frame" id="frame-140636034196976">
  <h4>File <cite class="filename">"/home/lubuntu/cars_project/carsandbids_project/carsandbids/carsandbids/selenium_api.py"</cite>,
      line <em class="line">47</em>,
      in <code class="function">get_page</code></h4>
  <div class="source "><pre class="line before"><span class="ws">        </span>request.url = request.url.replace(&quot;offset=50&quot;,&#x27;offset=0&#x27;)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws"></span>def get_page(url):</pre>
<pre class="line before"><span class="ws">    </span># api_key = random.choice(scraper_apis)</pre>
<pre class="line before"><span class="ws">    </span># api_url = f&#x27;http://api.scraperapi.com?api_key={api_key}&amp;url={url}&#x27;</pre>
<pre class="line current"><span class="ws">    </span>driver = webdriver.Chrome(executable_path=&quot;/home/lubuntu/sublime/chromedriver&quot;,options=ch_options,seleniumwire_options=options)</pre>
<pre class="line after"><span class="ws">    </span>driver.get(url)</pre>
<pre class="line after"><span class="ws">    </span>try:</pre>
<pre class="line after"><span class="ws">        </span>element = WebDriverWait(driver,10).until(EC.visibility_of_element_located((By.CLASS_NAME,&quot;quick-facts&quot;)))</pre>
<pre class="line after"><span class="ws">    </span>except:</pre>
<pre class="line after"><span class="ws">        </span>pass</pre></div>
</div>

<li><div class="frame" id="frame-140636034197088">
  <h4>File <cite class="filename">"/home/lubuntu/.local/lib/python3.9/site-packages/seleniumwire/webdriver.py"</cite>,
      line <em class="line">211</em>,
      in <code class="function">__init__</code></h4>
  <div class="source "><pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>if seleniumwire_options.get(&#x27;auto_config&#x27;, True):</pre>
<pre class="line before"><span class="ws">            </span>for key, value in config.items():</pre>
<pre class="line before"><span class="ws">                </span>chrome_options.set_capability(key, value)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line current"><span class="ws">        </span>super().__init__(*args, **kwargs)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span>class Safari(InspectRequestsMixin, DriverCommonMixin, _Safari):</pre>
<pre class="line after"><span class="ws">    </span>&quot;&quot;&quot;Extend the Safari webdriver to provide additional methods for inspecting requests.&quot;&quot;&quot;</pre>
<pre class="line after"><span class="ws"></span> </pre></div>
</div>

<li><div class="frame" id="frame-140636034197200">
  <h4>File <cite class="filename">"/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/chrome/webdriver.py"</cite>,
      line <em class="line">70</em>,
      in <code class="function">__init__</code></h4>
  <div class="source "><pre class="line before"><span class="ws">        </span>else:</pre>
<pre class="line before"><span class="ws">            </span>keep_alive = True</pre>
<pre class="line before"><span class="ws">        </span>if not service:</pre>
<pre class="line before"><span class="ws">            </span>service = Service(executable_path, port, service_args, service_log_path)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line current"><span class="ws">        </span>super(WebDriver, self).__init__(DesiredCapabilities.CHROME[&#x27;browserName&#x27;], &quot;goog&quot;,</pre>
<pre class="line after"><span class="ws">                                        </span>port, options,</pre>
<pre class="line after"><span class="ws">                                        </span>service_args, desired_capabilities,</pre>
<pre class="line after"><span class="ws">                                        </span>service_log_path, service, keep_alive)</pre></div>
</div>

<li><div class="frame" id="frame-140636034197312">
  <h4>File <cite class="filename">"/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/chromium/webdriver.py"</cite>,
      line <em class="line">90</em>,
      in <code class="function">__init__</code></h4>
  <div class="source "><pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>if not service:</pre>
<pre class="line before"><span class="ws">            </span>raise AttributeError(&#x27;service cannot be None&#x27;)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>self.service = service</pre>
<pre class="line current"><span class="ws">        </span>self.service.start()</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>try:</pre>
<pre class="line after"><span class="ws">            </span>RemoteWebDriver.__init__(</pre>
<pre class="line after"><span class="ws">                </span>self,</pre>
<pre class="line after"><span class="ws">                </span>command_executor=ChromiumRemoteConnection(</pre></div>
</div>

<li><div class="frame" id="frame-140636034197424">
  <h4>File <cite class="filename">"/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/common/service.py"</cite>,
      line <em class="line">81</em>,
      in <code class="function">start</code></h4>
  <div class="source "><pre class="line before"><span class="ws">                                            </span>creationflags=self.creationflags)</pre>
<pre class="line before"><span class="ws">        </span>except TypeError:</pre>
<pre class="line before"><span class="ws">            </span>raise</pre>
<pre class="line before"><span class="ws">        </span>except OSError as err:</pre>
<pre class="line before"><span class="ws">            </span>if err.errno == errno.ENOENT:</pre>
<pre class="line current"><span class="ws">                </span>raise WebDriverException(</pre>
<pre class="line after"><span class="ws">                    </span>&quot;&#x27;%s&#x27; executable needs to be in PATH. %s&quot; % (</pre>
<pre class="line after"><span class="ws">                        </span>os.path.basename(self.path), self.start_error_message)</pre>
<pre class="line after"><span class="ws">                </span>)</pre>
<pre class="line after"><span class="ws">            </span>elif err.errno == errno.EACCES:</pre>
<pre class="line after"><span class="ws">                </span>raise WebDriverException(</pre></div>
</div>
</ul>
  <blockquote>selenium.common.exceptions.WebDriverException: Message: &#x27;chromedriver&#x27; executable needs to be in PATH. Please see https://chromedriver.chromium.org/home

</blockquote>
</div>

<div class="plain">
    <p>
      This is the Copy/Paste friendly version of the traceback.
    </p>
    <textarea cols="50" rows="10" name="code" readonly>Traceback (most recent call last):
  File &quot;/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/common/service.py&quot;, line 71, in start
    self.process = subprocess.Popen(cmd, env=self.env,
  File &quot;/usr/lib/python3.9/subprocess.py&quot;, line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File &quot;/usr/lib/python3.9/subprocess.py&quot;, line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: &#x27;/home/lubuntu/sublime/chromedriver&#x27;

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File &quot;/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py&quot;, line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File &quot;/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py&quot;, line 2076, in wsgi_app
    response = self.handle_exception(e)
  File &quot;/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py&quot;, line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File &quot;/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py&quot;, line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &quot;/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py&quot;, line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File &quot;/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py&quot;, line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File &quot;/home/lubuntu/cars_project/carsandbids_project/carsandbids/carsandbids/selenium_api.py&quot;, line 107, in page
    return get_page(url)
  File &quot;/home/lubuntu/cars_project/carsandbids_project/carsandbids/carsandbids/selenium_api.py&quot;, line 47, in get_page
    driver = webdriver.Chrome(executable_path=&quot;/home/lubuntu/sublime/chromedriver&quot;,options=ch_options,seleniumwire_options=options)
  File &quot;/home/lubuntu/.local/lib/python3.9/site-packages/seleniumwire/webdriver.py&quot;, line 211, in __init__
    super().__init__(*args, **kwargs)
  File &quot;/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/chrome/webdriver.py&quot;, line 70, in __init__
    super(WebDriver, self).__init__(DesiredCapabilities.CHROME[&#x27;browserName&#x27;], &quot;goog&quot;,
  File &quot;/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/chromium/webdriver.py&quot;, line 90, in __init__
    self.service.start()
  File &quot;/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/common/service.py&quot;, line 81, in start
    raise WebDriverException(
selenium.common.exceptions.WebDriverException: Message: &#x27;chromedriver&#x27; executable needs to be in PATH. Please see https://chromedriver.chromium.org/home

</textarea>
</div>
<div class="explanation">
  The debugger caught an exception in your WSGI application.  You can now
  look at the traceback which led to the error.  <span class="nojavascript">
  If you enable JavaScript you can also use additional features such as code
  execution (if the evalex feature is enabled), automatic pasting of the
  exceptions and much more.</span>
</div>
      <div class="footer">
        Brought to you by <strong class="arthur">DON'T PANIC</strong>, your
        friendly Werkzeug powered traceback interpreter.
      </div>
    </div>

    <div class="pin-prompt">
      <div class="inner">
        <h3>Console Locked</h3>
        <p>
          The console is locked and needs to be unlocked by entering the PIN.
          You can find the PIN printed out on the standard output of your
          shell that runs the server.
        <form>
          <p>PIN:
            <input type=text name=pin size=14>
            <input type=submit name=btn value="Confirm Pin">
        </form>
      </div>
    </div>
  </body>
</html>

<!--

Traceback (most recent call last):
  File "/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/common/service.py", line 71, in start
    self.process = subprocess.Popen(cmd, env=self.env,
  File "/usr/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/home/lubuntu/sublime/chromedriver'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py", line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py", line 2076, in wsgi_app
    response = self.handle_exception(e)
  File "/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/lubuntu/.local/lib/python3.9/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/lubuntu/cars_project/carsandbids_project/carsandbids/carsandbids/selenium_api.py", line 107, in page
    return get_page(url)
  File "/home/lubuntu/cars_project/carsandbids_project/carsandbids/carsandbids/selenium_api.py", line 47, in get_page
    driver = webdriver.Chrome(executable_path="/home/lubuntu/sublime/chromedriver",options=ch_options,seleniumwire_options=options)
  File "/home/lubuntu/.local/lib/python3.9/site-packages/seleniumwire/webdriver.py", line 211, in __init__
    super().__init__(*args, **kwargs)
  File "/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/chrome/webdriver.py", line 70, in __init__
    super(WebDriver, self).__init__(DesiredCapabilities.CHROME['browserName'], "goog",
  File "/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/chromium/webdriver.py", line 90, in __init__
    self.service.start()
  File "/home/lubuntu/.local/lib/python3.9/site-packages/selenium/webdriver/common/service.py", line 81, in start
    raise WebDriverException(
selenium.common.exceptions.WebDriverException: Message: 'chromedriver' executable needs to be in PATH. Please see https://chromedriver.chromium.org/home



-->
